storm-core/src/jvm/backtype/storm/spout/CheckPointState.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.spout;
+
+/**
+ * Captures the current state of the transaction in
+ * {@link CheckpointSpout}
+ */
+public class CheckPointState {
+    public long txid;
+    public State state;
+
+    public enum State {
+        /**
+         * The checkpoint spout has committed the transaction.
+         */
+        COMMITTED,
+        /**
+         * The checkpoint spout has started committing the transaction.
+         */
+        COMMITTING,
```

Comment:
Named it COMMITTING and PREPARING since its short. We could probably have it as COMMIT_IN_PROGRESS and PREPARE_IN_PROGRESS. Prefer to keep it short and can add more comments.